cmd/pessimism/main.go

```diff
@@ -72,5 +107,28 @@ func main() {
 	logger.Debug("Waiting for all application threads to end")
 
 	logger.Info("Successful pessimism shutdown")
-	os.Exit(0)
+	return nil
+}
+
+func initializeMetrics(ctx context.Context, cfg *config.Config) *metrics.Metrics {
+	logger := logging.WithContext(ctx)
+
+	met := metrics.NewMetrics()
```

Comment:
Let's only construct the object if metrics are enabled to avoid erroneous memory allocations